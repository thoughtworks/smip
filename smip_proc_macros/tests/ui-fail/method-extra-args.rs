use smip::*;

#[service(id = 0x1234, major_version = 1, minor_version = 0)]
struct Service;

#[methods_impl]
impl Service {
    #[smip_method(id = 1)]
    fn method_with_more_than_one_arg(&self, arg1: u32, arg2: String) {}
}


fn main() {}