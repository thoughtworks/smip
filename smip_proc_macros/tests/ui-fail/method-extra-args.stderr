error: method should have exactly one arguments
 --> tests/ui-fail/method-extra-args.rs:9:38
  |
9 |     fn method_with_more_than_one_arg(&self, arg1: u32, arg2: String) {}
  |                                      ^
